From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Klop233 <3337913379@qq.com>
Date: Thu, 22 Dec 2022 17:19:57 +0800
Subject: [PATCH] Removed EULA spam


diff --git a/src/main/java/net/minecraft/server/Main.java b/src/main/java/net/minecraft/server/Main.java
index 110d11dfe7fbe64123800a82c3e8340f757d285b..a6a5a52a4c37f4b9aeda8b3274540ae2fb5d45b5 100644
--- a/src/main/java/net/minecraft/server/Main.java
+++ b/src/main/java/net/minecraft/server/Main.java
@@ -121,12 +121,6 @@ public class Main {
 
             // Spigot Start
             boolean eulaAgreed = true;
-            if ( eulaAgreed )
-            {
-                System.err.println( "You have used the Spigot command line EULA agreement flag." );
-                System.err.println( "By using this setting you are indicating your agreement to Mojang's EULA (https://account.mojang.com/documents/minecraft_eula)." );
-                System.err.println( "If you do not agree to the above EULA please stop your server and remove this flag immediately." );
-            }
             // Spigot End
             if (!eula.a() && !eulaAgreed) { // Spigot
                 Main.LOGGER.info("You need to agree to the EULA in order to run the server. Go to eula.txt for more info.");
