From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Klop233 <3337913379@qq.com>
Date: Thu, 22 Dec 2022 17:21:34 +0800
Subject: [PATCH] Cleaned code


diff --git a/src/main/java/net/minecraft/server/Main.java b/src/main/java/net/minecraft/server/Main.java
index a6a5a52a4c37f4b9aeda8b3274540ae2fb5d45b5..645ad9d10ff788cba47907fc5e16b2bf9a8a4083 100644
--- a/src/main/java/net/minecraft/server/Main.java
+++ b/src/main/java/net/minecraft/server/Main.java
@@ -112,21 +112,12 @@ public class Main {
             // Paper end
 
             java.nio.file.Path java_nio_file_path1 = Paths.get("eula.txt");
-            EULA eula = new EULA(java_nio_file_path1);
 
             if (optionset.has("initSettings")) { // CraftBukkit
                 Main.LOGGER.info("Initialized '{}' and '{}'", java_nio_file_path.toAbsolutePath(), java_nio_file_path1.toAbsolutePath());
                 return;
             }
 
-            // Spigot Start
-            boolean eulaAgreed = true;
-            // Spigot End
-            if (!eula.a() && !eulaAgreed) { // Spigot
-                Main.LOGGER.info("You need to agree to the EULA in order to run the server. Go to eula.txt for more info.");
-                return;
-            }
-
             org.spigotmc.SpigotConfig.disabledAdvancements = spigotConfiguration.getStringList("advancements.disabled"); // Paper - fix SPIGOT-5885, must be set early in init
             // Paper start - fix SPIGOT-5824
             File file;
