From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Klop233 <3337913379@qq.com>
Date: Thu, 22 Dec 2022 17:29:48 +0800
Subject: [PATCH] Localized some content


diff --git a/src/main/java/org/bukkit/craftbukkit/Main.java b/src/main/java/org/bukkit/craftbukkit/Main.java
index a44f0cadc3c304b38a86b647e4070c44f5f83d1e..516d272e97fd2931e518f05b8e3da2983591c473 100644
--- a/src/main/java/org/bukkit/craftbukkit/Main.java
+++ b/src/main/java/org/bukkit/craftbukkit/Main.java
@@ -283,13 +283,13 @@ public class Main {
                     String javaInfo = "Java " + runtimeMX.getSpecVersion() + " (" + runtimeMX.getVmName() + " " + runtimeMX.getVmVersion() + ")";
                     String osInfo = "Host: " + osMX.getName() + " " + osMX.getVersion() + " (" + osMX.getArch() + ")";
 
-                    System.out.println("System Info: " + javaInfo + " " + osInfo);
+                    System.out.println("系统信息: " + javaInfo + " " + osInfo);
                 } else {
                     System.out.println("Unable to read system info");
                 }
                 // Paper end
                 System.setProperty( "library.jansi.version", "Paper" ); // Paper - set meaningless jansi version to prevent git builds from crashing on Windows
-                System.out.println("Loading libraries, please wait...");
+                System.out.println("正在加载依赖库....");
                 net.minecraft.server.Main.main(options);
             } catch (Throwable t) {
                 t.printStackTrace();
diff --git a/src/main/java/org/spigotmc/WatchdogThread.java b/src/main/java/org/spigotmc/WatchdogThread.java
index b650f9b6901f15b7fec7a426cd77660039eecd66..7fcbdd139c6a5f72ddcaf14716bcb4a2cbd50818 100644
--- a/src/main/java/org/spigotmc/WatchdogThread.java
+++ b/src/main/java/org/spigotmc/WatchdogThread.java
@@ -186,12 +186,12 @@ public class WatchdogThread extends Thread
                 // Paper end
                 } else
                 {
-                    log.log(Level.SEVERE, "--- DO NOT REPORT THIS TO PAPER - THIS IS NOT A BUG OR A CRASH  - " + Bukkit.getServer().getVersion() + " ---");
-                    log.log(Level.SEVERE, "The server has not responded for " + (currentTime - lastTick) / 1000 + " seconds! Creating thread dump");
+                    log.log(Level.SEVERE, "--- 不要向 StarTuinity 提交这个问题 - 这不是 Bug 或崩溃  - " + Bukkit.getServer().getVersion() + " ---");
+                    log.log(Level.SEVERE, "服务端主线程已经 " + (currentTime - lastTick) / 1000 + " 秒没有响应了! 创建线程 Dump");
                 }
                 // Paper end - Different message for short timeout
                 log.log( Level.SEVERE, "------------------------------" );
-                log.log( Level.SEVERE, "Server thread dump (Look for plugins here before reporting to Paper!):" ); // Paper
+                log.log( Level.SEVERE, "请检查你的插件有没有阻塞主线程，或服务器负载过高:" ); // Paper
                 ChunkTaskManager.dumpAllChunkLoadInfo(); // Paper
                 this.dumpTickingInfo(); // Tuinity - log detailed tick information
                 dumpThread( ManagementFactory.getThreadMXBean().getThreadInfo( server.serverThread.getId(), Integer.MAX_VALUE ), log );
@@ -207,7 +207,7 @@ public class WatchdogThread extends Thread
                     dumpThread( thread, log );
                 }
                 } else {
-                    log.log(Level.SEVERE, "--- DO NOT REPORT THIS TO PAPER - THIS IS NOT A BUG OR A CRASH ---");
+                    log.log(Level.SEVERE, "--- DO NOT REPORT THIS TO STAR-TUINITY - THIS IS NOT A BUG OR A CRASH ---");
                 }
 
 
