From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: wangxyper <wangxyper@163.com>
Date: Thu, 22 Dec 2022 20:29:01 +0800
Subject: [PATCH] Try fix a IOOBE in ActivationRange


diff --git a/src/main/java/org/spigotmc/ActivationRange.java b/src/main/java/org/spigotmc/ActivationRange.java
index e1a0d9a83288b79064b5c9fd0b77082a48a7371e..787809b8af31f01d892bebfab3303f7ae33de1c7 100644
--- a/src/main/java/org/spigotmc/ActivationRange.java
+++ b/src/main/java/org/spigotmc/ActivationRange.java
@@ -254,6 +254,10 @@ public class ActivationRange
      */
     public synchronized static int checkEntityImmunities(Entity entity) // Paper - return # of ticks to get immunity
     {
+        if (entity == null){
+            return 1;
+        }
+
         // Paper start
         SpigotWorldConfig config = entity.world.spigotConfig;
         int inactiveWakeUpImmunity = checkInactiveWakeup(entity);
