From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Klop233 <3337913379@qq.com>
Date: Thu, 22 Dec 2022 17:46:30 +0800
Subject: [PATCH] Localized some content and removed some annoying spam


diff --git a/src/main/java/net/minecraft/server/dedicated/DedicatedServer.java b/src/main/java/net/minecraft/server/dedicated/DedicatedServer.java
index 3713a110a64fa686e785b9789c33dd09cacc2f48..15ab27896cb39900da66404d44e7fa26b314deed 100644
--- a/src/main/java/net/minecraft/server/dedicated/DedicatedServer.java
+++ b/src/main/java/net/minecraft/server/dedicated/DedicatedServer.java
@@ -254,14 +254,14 @@ public class DedicatedServer extends MinecraftServer implements IMinecraftServer
         // Paper end
 
         this.P();
-        DedicatedServer.LOGGER.info("Starting Minecraft server on {}:{}", this.getServerIp().isEmpty() ? "*" : this.getServerIp(), this.getPort());
+        DedicatedServer.LOGGER.info("正在在 {}:{} 上启动 Minecraft 服务器", this.getServerIp().isEmpty() ? "*" : this.getServerIp(), this.getPort());
 
         try {
             this.getServerConnection().bind(bindAddress); // Paper - Unix domain socket support
         } catch (IOException ioexception) {
-            DedicatedServer.LOGGER.warn("**** FAILED TO BIND TO PORT!");
+            DedicatedServer.LOGGER.warn("---  绑定端口失败 ---");
             DedicatedServer.LOGGER.warn("The exception was: {}", ioexception.toString());
-            DedicatedServer.LOGGER.warn("Perhaps a server is already running on that port?");
+            DedicatedServer.LOGGER.warn("最可能的原因: 你设置的端口已被占用");
             return false;
         }
 
@@ -272,8 +272,7 @@ public class DedicatedServer extends MinecraftServer implements IMinecraftServer
         // CraftBukkit end
 
         if (!this.getOnlineMode()) {
-            DedicatedServer.LOGGER.warn("**** SERVER IS RUNNING IN OFFLINE/INSECURE MODE!");
-            DedicatedServer.LOGGER.warn("The server will make no attempt to authenticate usernames. Beware.");
+            DedicatedServer.LOGGER.warn("使用离线模式启动服务器");
             // Spigot start
             if (org.spigotmc.SpigotConfig.bungee) {
                 DedicatedServer.LOGGER.warn("Whilst this makes it possible to use BungeeCord, unless access to your server is properly restricted, it also opens up the ability for hackers to connect with any username they choose.");
@@ -282,7 +281,7 @@ public class DedicatedServer extends MinecraftServer implements IMinecraftServer
                 DedicatedServer.LOGGER.warn("While this makes the game possible to play without internet access, it also opens up the ability for hackers to connect with any username they choose.");
             }
             // Spigot end
-            DedicatedServer.LOGGER.warn("To change this, set \"online-mode\" to \"true\" in the server.properties file.");
+            // DedicatedServer.LOGGER.warn("To change this, set \"online-mode\" to \"true\" in the server.properties file.");
         }
 
         if (!NameReferencingFileConverter.e(this)) {
