From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Klop233 <3337913379@qq.com>
Date: Thu, 22 Dec 2022 18:02:24 +0800
Subject: [PATCH] Localized some content


diff --git a/src/main/java/net/minecraft/server/MinecraftServer.java b/src/main/java/net/minecraft/server/MinecraftServer.java
index fcc9ad320f3f707057b18caa7fb49f643a7216bf..5c5ca2778071c8ce826d35f69298d2609ba852fd 100644
--- a/src/main/java/net/minecraft/server/MinecraftServer.java
+++ b/src/main/java/net/minecraft/server/MinecraftServer.java
@@ -910,7 +910,7 @@ public abstract class MinecraftServer extends IAsyncTaskHandlerReentrant<TickTas
         }
         // Paper end
         // CraftBukkit end
-        MinecraftServer.LOGGER.info("Stopping server");
+        MinecraftServer.LOGGER.info("*** Stopping Server | 正在停止服务器  ***");
         MinecraftTimings.stopServer(); // Paper
         // CraftBukkit start
         if (this.server != null) {
@@ -923,13 +923,13 @@ public abstract class MinecraftServer extends IAsyncTaskHandlerReentrant<TickTas
         }
 
         if (this.playerList != null) {
-            MinecraftServer.LOGGER.info("Saving players");
+            MinecraftServer.LOGGER.info("正在保存玩家数据");
             this.playerList.savePlayers();
             this.playerList.shutdown(this.isRestarting); // Paper
             try { Thread.sleep(100); } catch (InterruptedException ex) {} // CraftBukkit - SPIGOT-625 - give server at least a chance to send packets
         }
 
-        MinecraftServer.LOGGER.info("Saving worlds");
+        MinecraftServer.LOGGER.info("正在保存世界");
         Iterator iterator = this.getWorlds().iterator();
 
         WorldServer worldserver;
diff --git a/src/main/java/net/minecraft/server/level/PlayerChunkMap.java b/src/main/java/net/minecraft/server/level/PlayerChunkMap.java
index d827e18d8c3b9b5869dbb1e233f415ba0efb7c1b..0b831167d2930bf694052591629f08bc1c80f715 100644
--- a/src/main/java/net/minecraft/server/level/PlayerChunkMap.java
+++ b/src/main/java/net/minecraft/server/level/PlayerChunkMap.java
@@ -998,7 +998,7 @@ public class PlayerChunkMap extends IChunkLoader implements PlayerChunk.d {
             });
             this.world.asyncChunkTaskManager.flush(); // Paper - flush to preserve behavior compat with pre-async behaviour
 //            this.i(); // Paper - nuke IOWorker
-            PlayerChunkMap.LOGGER.info("ThreadedAnvilChunkStorage ({}): All chunks are saved", this.w.getName());
+            PlayerChunkMap.LOGGER.info("ThreadedAnvilChunkStorage ({}): 所有区块已保存", this.w.getName());
         } else {
             visibleChunks.values().stream().filter(PlayerChunk::hasBeenLoaded).forEach((playerchunk) -> {
                 IChunkAccess ichunkaccess = (IChunkAccess) playerchunk.getChunkSave().getNow(null); // CraftBukkit - decompile error
